oauth2_http/java/com/google/auth/oauth2/UserCredentials.java

```diff
@@ -124,13 +124,15 @@ static UserCredentials fromJson(Map<String, Object> json, HttpTransportFactory t
           "Error reading user credential from JSON, "
               + " expecting 'client_id', 'client_secret' and 'refresh_token'.");
     }
+    String tokenUrl = (String) json.get("token_uri");
+    URI tokenUri = (tokenUrl == null || tokenUrl.isEmpty()) ? null : URI.create(tokenUrl);
```

Comment:
qq, is this a value that exists? I was looking at my credentials.json file in the well known ADC path and I don't think I see an entry for tokenUri. Perhaps I needed additional flags for `gcloud application-default login`?

Reply:
I don't think this exists in the standard credentials.json file. Rather iiuc, we are providing a workaround that if user edit their json file and add this field, we use value from it. (ref: http://b/382442857#comment73)

Reply:
Can we add that as a quick comment above? I'm not entirely sure that this is going to be permanent/ how long we intend to support this. 

And given the prior stuff about malicious URIs, perhaps a warning about modifying the token_uri/ using a JSON with a malicious token_uri?

Reply:
Good point. Although in this case, user has to explicitly provide this. @sai-sunder-s do we need to add similar warning here?

Reply:
I do not think we need a warning here because even if url is malicious, we will be sending it a token that is built purely using the data in the json.